tests/test_emulator.cpp

```diff
@@ -0,0 +1,314 @@
+/*
```

Comment:
My plans are implementing all instructions currently supported, plus the ones needed for my guest kernel. Addind tests for all of those will take around 1000 LoC, so I won't split anything.

A full test suite for the entire x86 instruction set will certainly be much larger, so at some point it will need to be split. I'll consider it future work.